syntax = "proto3";

message JoinRequest {
    string name = 1;
    string host_port = 2;
}

message Res {
    bool ok = 1;
}

service Boss {
    rpc Join (JoinRequest) returns (Res);
}

message Next {
    string next_name = 1;
    string next_host_port = 2;
}

message Task {
    string from_name = 1;
}

service Worker {
    rpc SetNext (Next) returns (Res);
    rpc AcceptTask (Task) returns (Res);
}